### PR TITLE
#21: Add IntegrationTest for lastSyncTimestamp.

### PR DESCRIPTION
**Pull request by heshamMassoud**
- Add integration tests for last sync timestamp implementation.
- Refactor test logic for reuse.

### CODE REVIEW

File: src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java [hunk 1]
```diff
@@ -112,111 +170,259 @@ void run_WithSyncAsArgumentWithAllArg_ShouldExecuteAllSyncers() {
...
+  private void assertLastSyncCustomObjectIsCorrect(
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final String sourceProjectKey,
+      @Nonnull final String syncModule,
+      @Nonnull final Class<? extends BaseSyncStatistics> statisticsClass,
+      @Nonnull final ZonedDateTime lastSyncTimestamp) {
+
+    final CustomObjectQuery<LastSyncCustomObject> lastSyncQuery =
+        CustomObjectQuery.of(LastSyncCustomObject.class)
+            .byContainer(format("commercetools-project-sync.%s", syncModule));
```
Comment (ahmetoz) [inline src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java]: I would use APPLICATION_DEFAULT_NAME for this one. also "commercetools-project-sync.%s" is duplicated 3 times in same file

File: src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java [hunk 2]
```diff
@@ -48,23 +57,74 @@
...
   void setup() {
     syncerTestLogger.clearAll();
     cliRunnerTestLogger.clearAll();
     cleanUpProjects(createClient(CTP_SOURCE_CLIENT_CONFIG), createClient(CTP_TARGET_CLIENT_CONFIG));
+    setupSourceProjectData(createClient(CTP_SOURCE_CLIENT_CONFIG));
+  }
+
+  static void setupSourceProjectData(@Nonnull final SphereClient sourceProjectClient) {
+    final ProductTypeDraft productTypeDraft =
+        ProductTypeDraftBuilder.of(RESOURCE_KEY, "bar", "desc", emptyList()).build();
```
Comment (ahmetoz) [inline src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java]: in general this sample values like "foo", "bar" is not helpful (imo)

File: src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java (hunk 1 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java]: 4d00341

File: src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java (hunk 2 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java]: f028a4f

**Pull request merged**